### PR TITLE
Support do not disturb setting for platforms

### PR DESCRIPTION
**Pull request by Half-Shot**
Paired with https://github.com/vector-im/element-web/pull/14924 and https://github.com/vector-im/element-desktop/pull/133

### CODE REVIEW

**Comment by t3chguy** [comment]
This'll conflict with #5089

**Comment by jryans** [comment]
I'd like to get a Product call on this before proceeding with code review. Removing from queue for now.

**Comment by Half-Shot** [comment]
The failing test seems unrelated to my change?

File: src/Notifier.ts [hunk 1]
```diff
@@ -271,6 +273,7 @@ export const Notifier = {
         if (!plaf.supportsNotifications()) return false;
         if (!plaf.maySendNotifications()) return false;
 
+
```
Comment (turt2live) [inline src/Notifier.ts:276]: this is not needed

File: src/Notifier.ts [hunk 2]
```diff
@@ -79,6 +79,8 @@ export const Notifier = {
         if (!plaf.supportsNotifications() || !plaf.maySendNotifications()) {
             return;
         }
+        if (!plaf) return;
```
Comment (turt2live) [inline src/Notifier.ts:82]: this is after we already try to use it?

File: src/BasePlatform.ts [hunk 3]
```diff
@@ -142,6 +142,15 @@ export default abstract class BasePlatform {
         return false;
     }
 
+    /**
+     * Returns true if the application currently has permission
+     * to display notifications. Otherwise false.
```
Comment (turt2live) [inline src/BasePlatform.ts:147]: the booleans are in the wrong order, which is why we try to name everything with positive intent. Can we rename this to `isDisturbable` or something for positive intent?

**Review by turt2live — CHANGES_REQUESTED** [review]

**Comment by turt2live** [comment]
@Half-Shot is this still on your radar?

**Comment by Half-Shot** [comment]
As per the other PR, no :(

**Pull request closed without merging**